client/utils/auth.py

```diff
@@ -215,15 +226,26 @@ def log_message(self, format, *args):
             return
 
     server_address = (host_name, port_number)
-    httpd = http.server.HTTPServer(server_address, CodeHandler)
-    httpd.handle_request()
 
-    update_storage(access_token, expires_in, refresh_token)
-    return access_token
+    try:
+        httpd = http.server.HTTPServer(server_address, CodeHandler)
+        httpd.handle_request()
+    except OSError as e:
+        log.warning("HTTP Server Err {}".format(server_address), exc_info=True)
+        raise
+
+    if not auth_error:
+        update_storage(access_token, expires_in, refresh_token)
+        return access_token
+    else:
+        print("Authentication error: {}".format(auth_error))
```

Comment:
Handling the auth error also means that students can use OK without the token. 

The code always calls `auth.authenticate()` in some form to get an access token. Usually it's with `auth.authenticate(False)`.  If the second round of auth still fails that when the network protocols try to get it - then we will make a bunch of unnecessary requests - probably not the greatest idea. 

so TODO: explicitly handle bad auth.authenticate returning None in the protocols. I'll push that up in the next commit.